### ISSUE TITLE
Gulp, Redux - console says patching is complete but no changes seen

### ISSUE THREAD

**Issue by mikeebee** [milankinen/livereactload#111]
Hi there,

I'm attempting to retrofit this to an existing project that used Redux with React. According to the console, everything seems to be working, but I'm not seeing any changes.

```javascript
LiveReactload :: Connect reload client to ws://customurl.dev:4474
LiveReactload :: WebSocket client listening for changes...
```

Then, when I save a file I get...

```javascript
LiveReactload :: Bundle changed
LiveReactload :: Apply patch
```
Followed by numberous `> Patch module` message, finishing with... `LiveReactload :: Patching complete`

I am running the site through MAMP, so I'm not using `server.js` from the examples.

Here is the relevant parts from my Gulpfile.js

```javascript
var browserifyTask = function(options) {
    // Our app bundler
    var appBundler = browserify({
        entries: [options.src], // Only need initial file, browserify finds the rest
        transform: [ [babelify, {}], [envify, {}] ],
        plugin:  !options.development ? [] : [ lrload ],    // no additional configuration is needed
        debug: options.development, // Gives us sourcemapping
        cache: {},
        packageCache: {},
        fullPaths: options.development // Requirement of watchify
    });

    // The rebundle process
    var rebundle = function() {
        var start = Date.now();
        console.log('Building APP bundle');
        appBundler
            .bundle()
            .on('error', gutil.log)
            .pipe(source(inputOutput.jsOutput))
            .pipe(replace(/\[API_URL\]/g, API_URL))
            .pipe(gulpif(!options.development, streamify(uglify())))
            .pipe(gulp.dest(options.dest))
            .pipe(gulpif(options.development, livereload()))
            .pipe(notify(function() {
                console.log('APP bundle built in ' + (Date.now() - start) + 'ms');
            }));
    };

    // Fire up Watchify when developing
    if (options.development) {
        appBundler = watchify(appBundler);
        appBundler.on('update', rebundle);
    }

    rebundle();
};
```

I have the feeling I'm making a fundamental error, but I'm stumped! So any guidance will be appreciated.

Thanks

**Comment by alexmt** [comment]
Hello @mikeebee,

Looks like your Gulpfile does not have "livereactload/babel-transform" transformation plugin. Try to update babelify
transformation config as following:

```js
var appBundler = browserify({
    entries: [options.src], // Only need initial file, browserify finds the rest
    transform: [ [babelify, {
		  // Enable hot reloading using 'livereactload/babel-transform':
			plugins: options.development ? [
				['react-transform', { transforms: [{ transform: 'livereactload/babel-transform', imports: ['react'] }] }],
			] : []
		}], [envify, {}] ],
    plugin:  !options.development ? [] : [ lrload ],    // no additional configuration is needed
    debug: options.development, // Gives us sourcemapping
    cache: {},
    packageCache: {},
    fullPaths: options.development // Requirement of watchify
});
```

Thanks,
Alex

**Comment by mikeebee** [comment]
Hi @alexmt, thanks for getting back to me.

I already have this in my `.babelrc` file.

```json
{
  "presets": ["react", "es2015"],
  "env": {
    "development": {
      "plugins": [
        ["react-transform", {
          "transforms": [{
            "transform": "livereactload/babel-transform",
            "imports": ["react"]
          }]
        }],
    ],
      "ignore" : ["/js/src/libs/*", "/node_modules/*", "/js/src/plugins/*"]
    }
  }
}
```

Should that do the same thing? I tried adding what you suggested as well, but there is still no change.

Let me know if there is more information I can give you to help you, I know you're flying blind here!

**Comment by alexmt** [comment]
Hello @mikeebee,

I think you are right, ```.babelrc``` should do the same thing. My only guess is that NODE_ENV is not equals to 'development' and babel does not apply development options from ```.babelrc``` . I would suggest to try to move "plugins" into root node. e.g. 

```
{
  "presets": ["es2015", "react"],
  "plugins": [
    ["react-transform", {
      "transforms": [{
        "transform": "livereactload/babel-transform",
        "imports": ["react"]
      }]
    }]
  ],
 "ignore" : ["/js/src/libs/*", "/node_modules/*", "/js/src/plugins/*"]
}

```

**Comment by mikeebee** [comment]
I'm happy to have the settings in either place.

Unfortunately, I made the suggested `.babelrc` changes, but still no joy.

Silly Q but will it making any difference that there are a ton of non-react/redux JS files that get compiled along with everything? I notice that when I save one of those there is a `> Patch module ::` message for those too.

Is there any more information I can give you?

**Comment by milankinen** [comment]
Hmmm. Can you @mikeebee send the complete browser log you're receiving during the reload?

EDIT: and screencap from you js file stucture might help as well

**Comment by mikeebee** [comment]
Do you mean the console log? Here it is...

```javascript
chrome://net-internalsiled to get text for stylesheet 10536.44: No style sheet with given id found
VM4943:11 LiveReactload :: Connect reload client to ws://ballet.dev:4474
VM4943:11 LiveReactload :: WebSocket client listening for changes...
warning.js:47 Warning: Each child in an array or iterator should have a unique "key" prop. Check the render method of `Collection`. See https://fb.me/react-warning-keys for more information.warning @ warning.js:47
VM4943:11 LiveReactload :: Bundle changed
VM4943:11 LiveReactload :: Apply patch
VM4944:63  > Patch module    :: /Users/file-path-removed/js/src/react-apps/components/experienceSwitcher.js
VM4944:63  > Patch module    :: /Users/file-path-removed/js/src/react-apps/components/views/experience.js
VM4944:63  > Patch module    :: /Users/file-path-removed/js/src/react-apps/components/views/index.js
VM4944:63  > Patch module    :: /Users/file-path-removed/js/src/react-apps/events.js
VM4944:63  > Patch module    :: /Users/file-path-removed/js/src/react-apps/booking.js
VM4944:63  > Patch module    :: /Users/file-path-removed/js/src/react-apps/index.js
VM4944:63  > Patch module    :: /Users/file-path-removed/js/src/modules/drawerApps.js
VM4944:63  > Patch module    :: /Users/file-path-removed/js/src/react-apps/components/eventsItem.js
VM4944:63  > Patch module    :: /Users/file-path-removed/js/src/modules/bookingButtons.js
VM4944:63  > Patch module    :: /Users/file-path-removed/js/src/react-apps/components/eventsList.js
VM4944:63  > Patch module    :: /Users/file-path-removed/js/src/app.js
VM4944:63  > Patch module    :: /Users/file-path-removed/js/src/react-apps/components/views/events.js
VM4944:63  > Patch module    :: /Users/file-path-removed/js/src/react-apps/components/views/eventsList.js
VM4943:11 LiveReactload :: Patching complete
```
Here is my file list, I've also shown my store code as well just to make sure I'm not missing anything there!
![File list](http://mikeebee.com/other/file-structure.jpg)

**Comment by mikeebee** [comment]
@milankinen hey! You had a chance to look at this at all?

**Comment by milankinen** [comment]
Ahh sorry! For some reason I didn't get notification from this one. 😞  I'll take a look at this tonight.

**Comment by milankinen** [comment]
@mikeebee Based on your logs, it seems that you're trying to modify a component (I don't see any stores reloaded), is that correct? If that's so, this issue might have no relation to Redux. 

Are you using only pure function components? Would it be possible to see the sources (or at least exports and requires) of `experienceSwitcher.js`, `react-apps/index.js`, `app.js` and `eventsList.js` (or better, the whole project's sources)?

**Comment by mikeebee** [comment]
@milankinen no worries man! Yes, it's a component, I was just trying to make a markup change.

Pure function components... hmm, good question. I didn't write a lot of this project.

I can post some bits here but I think we'll come up short in the long run. Can you send me an email at mike@mikeebee.com and I'll send you a party pack.

**Comment by mikeebee** [comment]
Hey @milankinen, you ok to send me an email so I can send you some stuff?

**Comment by milankinen** [comment]
Sent

**Comment by mikeebee** [comment]
Thanks! Did you get my reply?

**Comment by mikeebee** [comment]
Do you think this could maybe something to do with me running through MAMP? I notice that if I go to `http://customurldev:4474` the browser markup says `Upgrade required`. I can't see how to specify the host.

**Comment by milankinen** [comment]
I've downloaded your app and started to investigate this. Unfortunately my master's thesis requires so much time that I have only little time to do OS at the moment. Sorry 😞

**Comment by mikeebee** [comment]
Not a problem at all! Whenever you get chance :)

**Comment by mikeebee** [comment]
Quick note, my components aren't pure functions, they are classes :)

Another thing, I did mention it before (but I'm not sure it's relevant), could you tell me how to specify the *host* in my Gulpfile please. I feel like me using MAMP is causing issues.

**Comment by milankinen** [comment]
The host should set to the hostname/IP of the machine *creating the bundle* (which run the gulp in this case).

I'll probably manage to finish the thesis writing during the next week so I'll be able to take a closer look at this.

**Comment by mikeebee** [comment]
Sorry, I meant I can't see how to actually set the host in Gulp but no worries, we can look at that when you get round to it. Good luck with the thesis!

**Comment by ilovepumpkin** [comment]
I have the exactly same issue.

**Comment by mikeebee** [comment]
It seems that my issues was caused by extending my classes, [see this comment](https://github.com/milankinen/livereactload/pull/118#issuecomment-236322622).

**Comment by milankinen** [comment]
Check. Please see the [proposed fix](https://github.com/milankinen/livereactload/pull/118#issuecomment-236858753)

**Comment by milankinen** [comment]
Fixed in `3.0.0`.
